Added support for resourcegroup and subscription configuraration assi…

KalpeshChavan12 opened the pull request:
…gnment.

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1308 


* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

msJinLei commented:
Please fix the compilation errors first.

It looks like the PR contains the breaking changes

```
Release the breaking changes
Breaking changes can only be release in major release, usually twice a year. Major releases are identified in the [Release Schedule](https://github.com/Azure/azure-powershell/milestones).

Please commit your breaking changes to main in the sprint of major release.

In the commit, you should

suppress the breaking change errors in "tools/StaticAnalysis/Exceptions/${ModuleName}/BreakingChangeIssues.csv"
"AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
"Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.NewAzFirewallPolicyNatRuleCommand","New-AzFirewallPolicyNatRule","0","1050","The parameter set '__AllParameterSets' for cmdlet 'New-AzFirewallPolicyNatRule' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'New-AzFirewallPolicyNatRule'."

remove the unnecessary breaking change notifications
``

msJinLei commented:
`Please note breaking change notification should be released 2 sprints ahead of the breaking change itself is released`
Your breaking change is able to include into the upcoming release and will target major release in fall.

KalpeshChavan12 commented:
@msJinLei PR wont contain any breaking changes, 

Only change in existing param is becomes optional.

As per doc, breaking change is if param become madatory.
https://github.com/Azure/azure-powershell/blob/main/documentation/breaking-changes/breaking-changes-attribute-help.md#cmdletparameterbreakingchangeattribute

msJinLei commented:
> @msJinLei PR wont contain any breaking changes,
> 
> Only change in existing param is becomes optional.
> 
> As per doc, breaking change is if param become madatory. https://github.com/Azure/azure-powershell/blob/main/documentation/breaking-changes/breaking-changes-attribute-help.md#cmdletparameterbreakingchangeattribute

It's good. Please fix the CI then.

KalpeshChavan12 commented:
@msJinLei We are waiting for .net sdk review, currently PR created and test recorded using local build.
once we publish .net SDK then CI will be fix.

It will be good if you review PR and provide feedback so changes can address ASAP

msJinLei commented:
> @msJinLei We are waiting for .net sdk review, currently PR created and test recorded using local build. once we publish .net SDK then CI will be fix.
> 
> It will be good if you review PR and provide feedback so changes can address ASAP

I find breaking changes in your PR and please remove the breaking change part in your PR. You can target them into next major release.
For other parts, please refer to the comments.

> **Please note the code freeze day is next Monday. For this release is very important and complicate, we are unprobable to merge PR after code freeze.  And can your PR can be ready before the code freeze? If you have strong business requirement. Please send email to Azure PowerShell team **

KalpeshChavan12 commented on src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationCreateOrUpdateMethod.cs:
```diff
@@ -267,16 +255,6 @@ public override void ExecuteCmdlet()
             HelpMessage = "Exclude KBs which require reboot")]
         public bool? WindowParameterExcludeKbRequiringReboot { get; set; }
 
-        [Parameter(
```
Added back pre/post task, since these are removed from contract, explicitly added message for deprecation

KalpeshChavan12 commented:
@msJinLei ,

 I have addressed all comments,

we want to release it in next month, currently date not fixed for release, we have complete PR review process, once release date finalized we will let you know about merging PR.

Please `DoNotMerge` PR

msJinLei commented:
> @msJinLei Lei Jin FTE ,
> 
> I have addressed all comments,
> 
> we want to release it in next month, currently date not fixed for release, we have complete PR review process, once release date finalized we will let you know about merging PR.
> 
> Please `DoNotMerge` PR

CC @vidai-msft

KalpeshChavan12 commented:
@msJinLei 
Any update?

vidai-msft commented:
@KalpeshChavan12 can you please fix the build error first?

KalpeshChavan12 commented:
@vidai-msft 
   It will be good have code review.
Build failed due to .Net SDK not yet available to nuget repo.
I have validate test recording and build locally using local feed.
.Net SDK has dependency with code swagger available to public repo. we will push swagger to public repo before release and we will unblock for build issue here.

It will be good if code get reviewed by team so it will be just CI/CD build fixing to get PR merge.

It will reduce time to merge PR.

KalpeshChavan12 commented:
> @KalpeshChavan12 can you please fix the build error first?

Can you please review PR.

Since our SDK not released and blocked due to release date. it will good if PR get reviewed ASAP

KalpeshChavan12 commented:
> > @msJinLei We are waiting for .net sdk review, currently PR created and test recorded using local build. once we publish .net SDK then CI will be fix.
> > It will be good if you review PR and provide feedback so changes can address ASAP
> 
> I find breaking changes in your PR and please remove the breaking change part in your PR. You can target them into next major release. For other parts, please refer to the comments.
> 
> > **Please note the code freeze day is next Monday. For this release is very important and complicate, we are unprobable to merge PR after code freeze.  And can your PR can be ready before the code freeze? If you have strong business requirement. Please send email to Azure PowerShell team **

Breaking changes are removed

vidai-msft commented:
@KalpeshChavan12 May I know when the .NET SDK will be ready?

vidai-msft commented:
@KalpeshChavan12 this PR still has issue and we have frozen code change for July release. I am going to clear the July milestone.

KalpeshChavan12 commented:
/azp run

KalpeshChavan12 commented:
> @KalpeshChavan12 this PR still has issue and we have frozen code change for July release. I am going to clear the July milestone.

Can you please help to trigger CI/CD checks for PR

vidai-msft commented:
/azu run

vidai-msft commented:
/azp run
